lib/grammers-client/src/client/auth.rs

```diff
@@ -7,20 +7,25 @@
 // except according to those terms.
 use super::net::connect_sender;
 use super::Client;
-use crate::types::LoginToken;
+use crate::types::{LoginToken, PasswordToken};
 use grammers_mtproto::mtp::RpcError;
 pub use grammers_mtsender::{AuthorizationError, InvocationError};
 use grammers_tl_types as tl;
 use std::convert::TryInto;
 use std::fmt;
 
+use grammers_crypto::two_factor_auth::{calculate_2fa, check_p_and_g};
+
 /// The error type which is returned when signing in fails.
 #[derive(Debug)]
 pub enum SignInError {
     SignUpRequired {
         terms_of_service: Option<tl::types::help::TermsOfService>,
     },
+    PasswordRequired(PasswordToken),
     InvalidCode,
+    InvalidParameters,
```

Comment:
We've decided this variant won't be exposed to the user and we'll panic instead so there's no need to have that here.

Reply:
Sorry, just forgot to remove this one